### PR TITLE
Code Health: Warnings

### PR DESCRIPTION
**Pull request by volodymyr-chekyrta**
This is the first PR in a series designed to enhance code quality
- Got rid of some warnings.
- Moved from using the android.defaults.buildfeatures.buildconfig property to setting build features at the module level.

| Before | After |
| ------------- | ------------- |
| Screenshot 2024-03-22 at 16 06 49| Screenshot 2024-03-22 at 16 22 38 |

Other issues will be addressed in a separate PR.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @volodymyr-chekyrta! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by omerhabib26** [comment]
Reviewing

**Review by k1rill — APPROVED** [review]
LGTM

**Review by omerhabib26 — APPROVED** [review]
only some minor nits

File: gradle.properties [hunk 1]
```diff
@@ -21,5 +21,4 @@ kotlin.code.style=official
 # resources declared in the library itself and none from the library's dependencies,
 # thereby reducing the size of the R class for that library
 android.nonTransitiveRClass=true
-android.defaults.buildfeatures.buildconfig=true
 android.nonFinalResIds=false
```
[the remark this answers is not in the document]
Comment (volodymyr-chekyrta) [reply gradle.properties]: Done ✅

File: discussion/src/main/java/org/openedx/discussion/presentation/ui/DiscussionUI.kt [hunk 2]
```diff
@@ -395,7 +416,10 @@ fun CommentMainItem(
                         .error(org.openedx.core.R.drawable.core_ic_default_profile_picture)
                         .placeholder(org.openedx.core.R.drawable.core_ic_default_profile_picture)
                         .build(),
-                    contentDescription = stringResource(id = org.openedx.core.R.string.core_accessibility_user_profile_image, comment.author),
+                    contentDescription = stringResource(
+                        id = org.openedx.core.R.string.core_accessibility_user_profile_image,
```
[the remark this answers is not in the document]
Comment (volodymyr-chekyrta) [reply discussion/src/main/java/org/openedx/discussion/presentation/ui/DiscussionUI.kt]: Done ✅

File: discussion/src/main/java/org/openedx/discussion/presentation/ui/DiscussionUI.kt [hunk 3]
```diff
@@ -94,7 +109,10 @@ fun ThreadMainItem(
                     .error(org.openedx.core.R.drawable.core_ic_default_profile_picture)
                     .placeholder(org.openedx.core.R.drawable.core_ic_default_profile_picture)
                     .build(),
-                contentDescription = stringResource(id = org.openedx.core.R.string.core_accessibility_user_profile_image, thread.author),
+                contentDescription = stringResource(
+                    id = org.openedx.core.R.string.core_accessibility_user_profile_image,
```
[the remark this answers is not in the document]
Comment (volodymyr-chekyrta) [reply discussion/src/main/java/org/openedx/discussion/presentation/ui/DiscussionUI.kt]: Done ✅

File: discussion/src/main/java/org/openedx/discussion/presentation/ui/DiscussionUI.kt [hunk 4]
```diff
@@ -245,7 +263,10 @@ fun CommentItem(
                         .error(org.openedx.core.R.drawable.core_ic_default_profile_picture)
                         .placeholder(org.openedx.core.R.drawable.core_ic_default_profile_picture)
                         .build(),
-                    contentDescription = stringResource(id = org.openedx.core.R.string.core_accessibility_user_profile_image, comment.author),
+                    contentDescription = stringResource(
+                        id = org.openedx.core.R.string.core_accessibility_user_profile_image,
```
[the remark this answers is not in the document]
Comment (volodymyr-chekyrta) [reply discussion/src/main/java/org/openedx/discussion/presentation/ui/DiscussionUI.kt]: Done ✅

**Comment by volodymyr-chekyrta** [comment]
@omerhabib26 PR is ready for another pass 🚀

**Review by omerhabib26 — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@volodymyr-chekyrta 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-app-android&entry.867055334=https://github.com/openedx/openedx-app-android/pull/267&entry.1484655318=https://github.com/volodymyr-chekyrta&entry.752974735=2024-03-22+14:31&entry.1917517419=2024-03-28+09:44&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**